src/shared/components/FragmentShareModal/FragmentShareModal.tsx

```diff
@@ -118,38 +120,24 @@ const FragmentShareModal: FC<FragmentShareModalProps & UserProps & EmbedFlowProp
 		setIsEmbedDropdownOpen(!isEmbedDropdownOpen);
 	};
 
-	const handleTabChanged = (id: string | number) => {
-		setActiveTab(id);
-		if (id === ShareDropdownTabs.EMBED) {
-			trackEvents(
-				{
-					object: item.external_id,
-					object_type: 'embed_code',
-					action: 'activate',
-				},
-				commonUser
-			);
-		}
-	};
-
 	useEffect(() => {
 		if (!isSmartSchoolEmbedFlow) {
 			setEmbedDropdownSelection('');
 		}
 	}, [isSmartSchoolEmbedFlow, tab]);
 
 	useEffect(() => {
-		if (isSmartSchoolEmbedFlow) {
+		if (isSmartSchoolEmbedFlow && isOpen) {
 			setActiveTab(ShareDropdownTabs.EMBED);
 			setEmbedDropdownSelection(EmbedCodeExternalWebsite.SMARTSCHOOL);
 		}
-	}, [isSmartSchoolEmbedFlow, setActiveTab, setEmbedDropdownSelection]);
+	}, [isSmartSchoolEmbedFlow, setActiveTab, setEmbedDropdownSelection, isOpen]);
 
 	useEffect(() => {
-		if (embedDropdownSelection === '') {
-			setEmbedCode(undefined);
-		} else {
-			setEmbedCode({
+		let newEmbedCode = undefined;
```

Comment:
prefer null over undefined when setting a variable yourself
for me it shows intent that you want this think to be null. and it isn't accidentally undefined.
but this is personal preference